### ISSUE TITLE
Pylance appends to star import for auto-imports not defined in `__all__`

### ISSUE THREAD

**Issue by NCPlayz** [microsoft/pylance-release#1679]
<h3> Environment data </h3>

-   Language Server version: 2021.8.1
-   OS and version: win32 x64
-   Python version (and distribution if applicable, e.g. Anaconda): 3.9.4 64-bit
-   python.analysis.indexing: undefined
-   python.analysis.typeCheckingMode: strict

<h3> Expected behaviour </h3>

A new import line is added for the specific symbol not defined in `__all__`:

```py
from test import *
from test import B
```

<h3> Actual behaviour </h3>

The symbol not defined in `__all__` is simply appended to the end of the star import:

```py
from test import *B,
```

<h3> Logs </h3>

Python Language Server Log

Typing "A" (see GIF):
```
Background analysis message: setFileOpened
Background analysis message: markFilesDirty
[FG] completion at test2.py:2:1 ...
[FG]   parsing: test2.py (0ms)
[FG]   binding: test2.py (0ms)
[FG] completion at test2.py:2:1 [found 89 items] (2ms)
Background analysis message: analyze
[BG(1)] analyzing: test2.py ...
[BG(1)]   parsing: test2.py (1ms)
[BG(1)]   binding: test2.py (0ms)
[BG(1)]   checking: test2.py (1ms)
[BG(1)] analyzing: test2.py (2ms)
Background analysis message: getDiagnosticsForRange
Background analysis message: resumeAnalysis
Background analysis message: getDiagnosticsForRange
Background analysis message: getSemanticTokens delta
[BG(1)] getSemanticTokens delta previousResultId:1629072695818 at test2.py (0ms)
Background analysis message: markFilesDirty
Background analysis message: analyze
Background analysis message: setFileOpened
Background analysis message: markFilesDirty
Background analysis message: analyze
[BG(1)] analyzing: test2.py ...
[BG(1)]   parsing: test2.py (0ms)
[BG(1)]   binding: test2.py (0ms)
[BG(1)]   checking: test2.py (0ms)
[BG(1)] analyzing: test2.py (0ms)
Background analysis message: resumeAnalysis
Background analysis message: getSemanticTokens delta
[BG(1)] getSemanticTokens delta previousResultId:1629072700448 at test2.py (2ms)
[FG] parsing: test2.py (0ms)
[FG] binding: test2.py (0ms)
Background analysis message: markFilesDirty
Background analysis message: analyze
```
Typing "B" and pressing enter for suggestion:
```
Background analysis message: setFileOpened
Background analysis message: markFilesDirty
[FG] completion at test2.py:2:1 ...
[FG]   parsing: test2.py (0ms)
[FG]   binding: test2.py (0ms)
[FG] completion at test2.py:2:1 [found 29 items] (1ms)
Background analysis message: getDiagnosticsForRange
Background analysis message: getDiagnosticsForRange
Background analysis message: getSemanticTokens delta
[BG(1)] getSemanticTokens delta previousResultId:1629072704015 at test2.py ...
[BG(1)]   parsing: test2.py (0ms)
[BG(1)]   binding: test2.py (1ms)
[BG(1)] getSemanticTokens delta previousResultId:1629072704015 at test2.py (3ms)
Background analysis message: analyze
[BG(1)] analyzing: test2.py ...
[BG(1)]   checking: test2.py (0ms)
[BG(1)] analyzing: test2.py (0ms)
Background analysis message: resumeAnalysis
Background analysis message: getDiagnosticsForRange
Background analysis message: getDiagnosticsForRange
Background analysis message: markFilesDirty
Background analysis message: analyze
Background analysis message: setFileOpened
Background analysis message: markFilesDirty
Background analysis message: setFileOpened
Background analysis message: markFilesDirty
Background analysis message: analyze
[BG(1)] analyzing: test2.py ...
[BG(1)]   parsing: test2.py (1ms)
[BG(1)]   binding: test2.py (0ms)
[BG(1)]   checking: test2.py (2ms)
[BG(1)] analyzing: test2.py (3ms)
Background analysis message: resumeAnalysis
Background analysis message: getSemanticTokens delta
[BG(1)] getSemanticTokens delta previousResultId:1629072758386 at test2.py (0ms)
[FG] parsing: test2.py (0ms)
[FG] binding: test2.py (0ms)
Background analysis message: getDiagnosticsForRange
Background analysis message: getDiagnosticsForRange
Background analysis message: setFileOpened
Background analysis message: markFilesDirty
Background analysis message: analyze
[BG(1)] analyzing: test2.py ...
[BG(1)]   parsing: test2.py (0ms)
[BG(1)]   binding: test2.py (0ms)
[BG(1)]   checking: test2.py (2ms)
[BG(1)] analyzing: test2.py (2ms)
Background analysis message: resumeAnalysis
Background analysis message: getSemanticTokens delta
[BG(1)] getSemanticTokens delta previousResultId:1629072760438 at test2.py (1ms)
[FG] parsing: test2.py (0ms)
[FG] binding: test2.py (0ms)
Background analysis message: markFilesDirty
Background analysis message: getDiagnosticsForRange
Background analysis message: getDiagnosticsForRange
Background analysis message: analyze
```

<h3> Code Snippet / Additional information </h3>

test.py
```python
__all__ = ("A",)


class A:
    ...


class B:
    ...
```

![HkhNVpobf6](https://user-images.githubusercontent.com/26068884/129497233-0d808c39-d4d3-44a6-9ebf-288ce16401d8.gif)

**Comment by AlonsoMackenlly** [comment]
The same problem with requests in python 3.8.9 after `python -m pip install requests`

![image](https://user-images.githubusercontent.com/19227850/131312687-cf29a8d1-fc96-4487-b4b9-cf0c03e4b7d0.png)

Also, none of the commented lines are recognized by the auto-import, I'm sure that these classes and functions are in the configured Python environment.

**Comment by NCPlayz** [comment]
This doesn't seem to be the same issue. I think this one just requires re-indexing the libraries? I usually reload window if this happens after installing a package and everything resolves fine.

**Comment by AlonsoMackenlly** [comment]
> This doesn't seem to be the same issue. I think this one just requires re-indexing the libraries? I usually reload window if this happens after installing a package and everything resolves fine.

Sorry, I wrote in the wrong topic, I made a mistake. No restarting does not help, I tried all possible options, manual import from django.. django.contrib.auth... it works, but it does not offer to import the existing library itself without manually entering paths

**Comment by AlonsoMackenlly** [comment]
> [ previous comment was deleted ]
> This doesn't seem to be the same issue. I think this one just requires re-indexing the libraries? I usually reload window if this happens after installing a package and everything resolves fine.

![image](https://user-images.githubusercontent.com/19227850/131313508-4c97c17a-27a5-40b7-9a1e-e10d2f95712d.png)

and, so

![image](https://user-images.githubusercontent.com/19227850/131313624-b23573c3-9acd-4f83-b55a-4ee0ee598305.png)

**Comment by jakebailey** [comment]
If you don't have indexing enabled, you may not get the right auto-import completions at all. Try setting `"python.analysis.indexing": true`.

**Comment by AlonsoMackenlly** [comment]
> If you don't have indexing enabled, you may not get the right auto-import completions at all. Try setting `"python.analysis.indexing": true`.

I tried to install this option, it is now enabled. I still can't get hints without importing this package in advance

**Comment by jakebailey** [comment]
This issue has been fixed in version 2021.9.4, which we've just released. You can find the changelog here: https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#202194-29-september-2021
